WorkloadManagementSystem/DB/JobElasticDB.py

```diff
@@ -0,0 +1,142 @@
+
+JOB_STATES = ['Submitting', 'Received', 'Checking', 'Staging', 'Waiting', 'Matched',
+              'Running', 'Stalled', 'Done', 'Completed', 'Failed']
+JOB_FINAL_STATES = ['Done', 'Completed', 'Failed']
+
+
+class JobDB(DB):
+
+  def __init__(self):
+    """ Standard Constructor
+    """
+
+    DB.__init__(self, 'jobelasticdb', 'WorkloadManagement/JobElasticDB')
+
+    self.jobAttributeNames = []
+
+    self.siteClient = SiteStatus()
+
+    self.log.info("==================================================")
+
+#############################################################################
+  def getJobParameters(self, jobID, paramList=None):
+    """ Get Job Parameters defined for jobID.
+      Returns a dictionary with the Job Parameters.
+      If parameterList is empty - all the parameters are returned.
+
+    :param self: self reference
+    :param int jobID: Job ID
+    :param list paramList: list of parameters to be returned
+    """
+
+    self.log.debug('JobDB.getParameters: Getting Parameters for job %s' % jobID)
+
+    resultDict = {}
+
+    if paramList:
+      paramNameList = []
+
+      for x in paramList:
+        paramNameList.append(x)
+
+      query = {
+          "query": {
+              "bool": {
+                  "must": [
+                      {"match": {"JobID": jobID}}, {"match": {"Name": ','.join(paramNameList)}}]}},
+          "_source": ["Name", "Value"]}
+
+    else:
+      query = {"query": {"match": {"JobID": jobID}}, "_source": ["Name", "Value"]}
+
+    gLogger.debug("Getting results for ", jobID)
+    result = self.query('jobelasticdb*', query)
+
+    if not result['OK']:
+      return S_ERROR(result)
+
+    sources = result['Value']['hits']['hits']
+
+    for source in sources:
+
+      name = source['_source']['Name']
+      value = source['_source']['Value']
+
+      try:
+        resultDict[name] = value.tostring()
+      except BaseException:
+        resultDict[name] = value
+
+    return S_OK(resultDict)
+
+#############################################################################
+  def setJobParameter(self, jobID, key, value):
+
+    """ Set a parameter specified by name,value pair for the job JobID
+
+    :param self: self reference
+    :param int jobID: Job ID
+    :param basestring key: Name
+    :param blob value: value
+    """
+
+    query = {
+        "query": {
+            "term": {
+                "JobID": jobID}}, "script": {
+            "inline": "ctx._source.Value = params.value; ctx._source.Name = params.name", "params": {
+                "value": value, "name": key}}}
+    indexName = self.generateFullIndexName('jobelasticdb')
+    result = self.exists(indexName)
+
+    if not result:
+      mapping = {
+          "JobParameters": {
+              "properties": {
+                  "JobID": {
+                      "type": "long"}, "Name": {
+                      "type": "text"}, "Value": {
+                      "type": "binary"}}}}
```

Comment:
binary is exists? The value can be number of string. I think keyword will be better: Value:{"type":"keyword"}

Reply:
Changed type to ```keyword```.